ONSdigital/ras-frontstage
# Handle missing/inactive IAC errors better

# Motivation and Context
If a user double clicks or went back then submitted again after signing up for a survey, we'd get a TypeError, some bad logging and an error screen.  This PR handles the error in a better way to make diagnosing the issue easier if needed (this is primarily a 1 time error).  It still shows an error page, but it's more intentional.  This logging is also not at exception level so it won't trigger alerts (as nothing is actually on fire when this happens)

# What has changed
- No more error/exception logging in this situation.  
- Better info log lines if we need to diagnose it for a respondent in the case of a real issue.
- A different error page shown to the user (previously the 500-error page, now the 400-error page).

# How to test?
- Run the acceptance tests to seed data
- Get an unused enrolment code that example@example.com user isn't signed up for from response-operations-ui
- Sign into frontstage and add survey.  You'll see the confirm page.  From here the error can be generated in 2 ways:
    - Double click the confirm button. Most of the time the first click will successfully add the user, the second click will fail because the IAC code now isn't active.
   - The second way is to click confirm once.  This will take you back to the todo page.  Press the back button in the browser.  It'll take you back to the confirm page.  Click confirm again and this'll trigger the error.

In the past there would be a typeError and a generic exception log line.  Now, the logging is more expressive, doesn't throw a log line with a level of error/exception (as nothing is on fire), and we intentionally render a 400 error page instead of falling back to the most generic error page we have.

# Links
https://trello.com/c/Xn1n6bAW

# Screenshots (if appropriate):
